packages/frontend/src/lib/pnl.ts

```diff
@@ -0,0 +1,95 @@
+import BigNumber from 'bignumber.js'
+
+import { TransactionType } from '@constants/enums'
+import { PositionType } from '../types'
+
+interface Deposit {
+  squeethAmount: BigNumber
+  ethAmount: BigNumber
+  usdValue: BigNumber
+  timestamp: string
+  transactionType: TransactionType
+  txId: string
+  ethPriceAtDeposit: BigNumber
+}
+
+interface LongDeposit extends Deposit {
+  sellQuote: BigNumber
+}
+
+interface GetCurrentLongParams {
+  positionType: string
+  squeethAmount: BigNumber
+  transactions: any[]
+  getSellQuote: (input: BigNumber) => Promise<Record<string, any>>
+}
+
+export async function getCurrentLong({
+  getSellQuote,
+  positionType,
+  squeethAmount,
+  transactions,
+}: GetCurrentLongParams) {
+  if (positionType === PositionType.SHORT) return []
+  let totalLongSqth = new BigNumber(0)
+
+  const result: LongDeposit[] = []
+  let sellQuote
+  for (let index = 0; index < transactions.length; index++) {
+    if (totalLongSqth.gte(squeethAmount)) break
```

Comment:
Do we just break the loop here, or we need to do smt else if the amount is greater than squth amount?